component/app.jsonnet

```diff
@@ -3,7 +3,41 @@ local inv = kap.inventory();
 local params = inv.parameters.openshift_prometheus_proxy;
 local argocd = import 'lib/argocd.libjsonnet';
 
-local app = argocd.App('openshift-prometheus-proxy', params.namespace);
+local app =
+  argocd.App('openshift-prometheus-proxy', params.namespace) {
+    spec+: {
+      ignoreDifferences+: [
```

Comment:
I'm not happy with having to do this, especially since the ignoreDifferences implementation currently only supports json pointers as defined by [RFC6901](https://tools.ietf.org/html/rfc6901), which don't support wildcard access to arrays.

See also ArgoCD issues https://github.com/argoproj/argo-cd/issues/5276 and https://github.com/argoproj/argo-cd/issues/4320

The root cause for having to do this is that OpenShift tends to update a resource's spec for many of the OpenShift specific resources (e.g. ImageStream and BuildConfig). And when using Deployments with ImageStreams and ImageStreamTags, OpenShift replaces image references to `latest` tags with the latest SHA256 of that image.

Reply:
Since the deployment manifests seem to be relatively simple, couldn't we just create them in the component (instead of reusing the template) and use K8s primitives?

Reply:
The biggest problem for that approach is that there's no image (or Dockerfile to build one) available for the proxy. It's built using an OpenShift S2I build in the cluster.